### PR TITLE
Docker utilities to help build and test on Linux

### PR DESCRIPTION
**Pull request by hartbit**

### CODE REVIEW

**Comment by hartbit** [comment]
@swift-ci please smoke test

File: utils/docker/docker-utils [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        "-t", "llbuild-docker-1604",
+        "--build-arg", "SNAPSHOT=%s" % snapshot_filename,
+        docker_dir
+    ])
+
+def run(args):
+    """Runs an executable in the container."""
+    docker_run([args.executable] + args.arguments)
+
+def bootstrap(args):
```
Comment (aciidb0mb3r) [inline utils/docker/docker-utils]: We should remove this, right?

**Comment by aciidb0mb3r** [comment]
We should add some documentation in Development.md on how to use these utilities.

**Comment by hartbit** [comment]
@aciidb0mb3r Is this documentation enough?

File: utils/docker/docker-utils (hunk 1 above)
Comment (hartbit) [reply utils/docker/docker-utils]: Yep, sorry. I removed some but missed those.

Comment (hartbit) [reply utils/docker/docker-utils]: Removed.

**Comment by neonichu** [comment]
I have a couple additional bits in my personal bits, so I'm just going to mention these here:

- `RUN update-alternatives --install /bin/sh sh /bin/bash 100` just to be safe w.r.t. accidental bash-isms that may have snuck in :)
- `RUN apt-get install -y vim` to have a sane editor inside the container
- `CMD ["/bin/bash"]` which avoids having to specify "bash" on when doing "run"

**Comment by hartbit** [comment]
@neonichu I've followed your suggestions. Thanks!

File: utils/docker/Dockerfile [hunk 2]
```diff
@@ -29,5 +31,4 @@ RUN tar -xvzf $SNAPSHOT --directory / --strip-components=1 && \
 # Set Swift Path
 ENV PATH /usr/bin:$PATH
 
-# Print Installed Swift Version
-RUN swift --version
+CMD ["/bin/bash"]
```
Comment (jakepetroules) [inline utils/docker/Dockerfile]: file is missing an ending newline

File: utils/docker/Dockerfile [hunk 3]
```diff
@@ -17,9 +17,11 @@ RUN apt-get -q update && \
     sqlite3 \
     python-pip \
     libsqlite3-dev \
-    libncurses5-dev
+    libncurses5-dev \
+    vim
```
Comment (jakepetroules) [inline utils/docker/Dockerfile]: seems unfair to arbitrarily choose editors to include in the image; if you want VIM you should install it manually

Comment (aciidb0mb3r) [reply utils/docker/Dockerfile]: I think it's fine to include vim as its bundled with major linux distributions anyway. Besides, I don't see it as a major issue.

Comment (jakepetroules) [reply utils/docker/Dockerfile]: I'd really prefer this not be here. There's no justification for it at all. Docker images are supposed to be kept as small as possible to do their job.

Comment (aciidb0mb3r) [reply utils/docker/Dockerfile]: I agree Docker images should be small but thats when deploying in production. This is really for just development convenience. I don't think it's unreasonable given our use case. Anyway, we can leave it out for now and add some way to customize such addons.

File: utils/docker/docker-utils [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+ -------------------------------------------------------------------------
+ This script contains helper operations to use Docker for SwitPM development
```
Comment (dmbryson) [inline utils/docker/docker-utils]: nit, but this is llbuild, not SwiftPM

File: utils/docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
```
Comment (dmbryson) [inline utils/docker/Dockerfile]: nit, all these copyright dates should be through 2018

**Comment by hartbit** [comment]
I fixed the comments and removed vim.

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by jakepetroules** [comment]
@hartbit In future please [squash and/or rebase into a series of atomic, logical commits](http://adopteungit.fr/en/methodology/2017/04/26/how-to-do-atomic-commits.html) as appropriate.

It's best to avoid having the intermediate states from code review (like "removed vim" or "fixed spelling error from previous commit") end up in master - this helps with bisecting and history readability.

**Comment by hartbit** [comment]
@jakepetroules will make sure to do so. Thanks!

**Pull request merged**